### DISCUSSION TITLE
It possible to have both repositories and definitions at the same time?

### DISCUSSION THREAD

**Discussion by dagsir** [dagster-io/dagster#17525] [Q&A] [answered]
hey, i have a bunch of ops, jobs, and graphs defined in a repositories.py file as dagster @repository decorated repos. i’d like to progressively migrate these to be dagster Definitions (using the software defined assets data model). is it possible to have both repositories and definitions at the same time?

---

*The question was originally asked in [Dagster Slack](https://dagster.slack.com/archives/C01U954MEER/p1698767008774119?thread_ts=1698767008.774119&cid=C01U954MEER).*

**Comment by yuhan** [comment, marked as answer]
They two can't co-exist. But the migration path from repository should be straightforward. Here are some examples: https://github.com/dagster-io/dagster/pull/11072/files and https://github.com/dagster-io/dagster/pull/11135/files. 

To further make the migration incremental, you can do it in steps:
1. migrate all objects defined in repository to definitions at once
2. keep using repository as the entry point for your deployment (so you don't need to change deployment set up yet in this step). you can convert a definitions object to a repository using [`create_repository_using_definitions_args`](https://docs.dagster.io/_apidocs/definitions#dagster.create_repository_using_definitions_args).
3. migrate the deployment setup from repository to definitions.
